File: Olympiad/COCI/official/2008/regional_solutions/platforme.pas
program platforme;

var
   N, i, rez : longint;
   platforme : array[1..100] of record y, x1, x2 : longint; end;

function spusti( x : double; y : longint ) : longint;
var
   i, rez : longint;
begin
   rez := 0;
   for i := 1 to N do begin
      if (platforme[i].y  < y) and
         (platforme[i].x1 < x) and
         (platforme[i].x2 > x) and
         (platforme[i].y  > rez) then
         rez := platforme[i].y;
   end;
   spusti := rez;
end;

begin
   readln(N);
   for i := 1 to N do
      readln(platforme[i].y, platforme[i].x1, platforme[i].x2);

   rez := 0;
   for i := 1 to N do begin
      rez := rez + platforme[i].y - spusti(platforme[i].x1 + 0.5, platforme[i].y);
      rez := rez + platforme[i].y - spusti(platforme[i].x2 - 0.5, platforme[i].y);
   end;
   writeln( rez );
end.

File: Olympiad/COCI/official/2008/regional_solutions/platforme.cpp
#include <stdio.h>

int N;
struct { int y, x1, x2; } platforme[100];

int spusti(double x, int y) {
   int rez = 0;
   for ( int i=0; i<N; ++i ) {
      if ( platforme[i].y  < y &&
           platforme[i].x1 < x &&
           platforme[i].x2 > x &&
           platforme[i].y  > rez ) {
         rez = platforme[i].y;
      }
   }
   return rez;
}

int main() {
   scanf( "%d", &N );
   for ( int i=0; i<N; ++i )
      scanf( "%d%d%d", &platforme[i].y, &platforme[i].x1, &platforme[i].x2 );

   int rez = 0;
   for ( int i=0; i<N; ++i ) {
      rez += platforme[i].y - spusti(platforme[i].x1 + 0.5, platforme[i].y);
      rez += platforme[i].y - spusti(platforme[i].x2 - 0.5, platforme[i].y);
   }
   printf( "%d\n", rez );

   return 0;
}
